include/nuttx/list.h

```diff
@@ -101,6 +101,17 @@
 #define list_last_entry(list, type, member) \
            list_entry((list)->prev, type, member)
 
+#define list_next_entry(list, type, member) \
```

Comment:
@xiaoxiang781216 despite I've just merged https://github.com/apache/nuttx/pull/10357 I have a doubts about it. Really it makes me feel that some of this macro definitions are copied from Linux.
In one of the comments I'm reading
> I have rewrite the code, and it is not same as any code, Both Linux and freebsd.

I found the Linux variant, but failed to find FreeBSD variant. It would be helpful if the link to FreeBSP file can be a part of a description.

I will post my thoughts here:
1. When I look into NuttX `list.h` back in history I see some small differences like NuttX variant use `for_every` while Linux variant use `for_each`. Those differences are small, but still seems to be significant. Also Linux variant use `list_entry` in many macro like `list_for_each_entry_continue` (like here) while NuttX variant use `container_of((list)->next, type, member)` directly (in `list_for_every_entry`).
2. In `bam43xxx` folder I see things like
```
  iframe = list_remove_head_type(&gbus->tx_queue,
                                 bcmf_interface_frame_t,
                                 list_entry);
...
struct list_node    list_entry;
...
list_add_tail(&ibus->rx_queue, &iframe->list_entry);
```
I also see that it includes `#include <nuttx/list.h>`, so I have two questions/observations: 
- Seems like `bcm43xxx` is not tested by CI
- How is this going to co-exist with `#define list_entry(ptr, type, member) container_of(ptr, type, member)`? I would expect pre-processor errors on code above.

Currently I do not see a real use-case for `list_entry` and `list_last_entry` based on above. I see that `list_first_entry` is used in `void task_uninit_info(FAR struct task_group_s *group)` so wee can keep it, but implement via `#define list_first_entry(list, type, member) container_of((list)->next, type, member)`.

What are your thoughts here @yamt @patacongo @jerpelea @acassis ?

Reply:
> I found the Linux variant, but failed to find FreeBSD variant. It would be helpful if the link to FreeBSP file can be a part of a description.

FreeBSD just copy Linux header file directly:
https://github.com/freebsd/freebsd-src/blob/main/sys/compat/linuxkpi/common/include/linux/list.h#L204

> 
> I also see that it includes `#include <nuttx/list.h>`, so I have two questions/observations:
> 

but, I can't find the inclusion, could you add the link?

> * Seems like `bcm43xxx` is not tested by CI

it enable in the following defconfig:
boards/arm/rp2040/raspberrypi-pico-w/configs/telnet/defconfig
boards/arm/stm32/emw3162/configs/wlan/defconfig
boards/arm/stm32/photon/configs/wlan/defconfig
boards/arm/stm32/photon/configs/wlan-perf/defconfig

> * How is this going to co-exist with `#define list_entry(ptr, type, member) container_of(ptr, type, member)`? I would expect pre-processor errors on code above.
> 
> Currently I do not see a real use-case for `list_entry` and `list_last_entry` based on above. I see that `list_first_entry` is used in `void task_uninit_info(FAR struct task_group_s *group)` so wee can keep it, but implement via `#define list_first_entry(list, type, member) container_of((list)->next, type, member)`.
> 

But it's natural to define list_entry/list_last_entry if list_first_entry is defined. My question is that is this mean we can't use any macro or function name from Linux?

> What are your thoughts here @yamt @patacongo @jerpelea @acassis ?

Reply:
> > I found the Linux variant, but failed to find FreeBSD variant. It would be helpful if the link to FreeBSP file can be a part of a description.
> 
> FreeBSD just copy Linux header file directly: https://github.com/freebsd/freebsd-src/blob/main/sys/compat/linuxkpi/common/include/linux/list.h#L204

are you sure?
i thought these "linuxkpi" stuff was about avoiding gpl pollution.